src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/SortedCaseCollection.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.service.impl;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.Optional;
+import java.util.TreeSet;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseDTO;
+
+/**
+ * Collection of CaseDTO objects, which may have been populated by RM or our local cache, and sorted
+ * by latest date first , where each object takes it's date in the following order:
+ *
+ * <ol>
+ *   <li>lastUpdated
+ *   <li>createdDateTime
+ * </ol>
+ */
+public class SortedCaseCollection {
+
+  static class CaseDateComparator implements Comparator<CaseDTO> {
+    @Override
+    public int compare(CaseDTO first, CaseDTO second) {
+      Date firstDate = dateForCompare(first);
+      Date secondDate = dateForCompare(second);
+      return firstDate.compareTo(secondDate);
+    }
+
+    private Date dateForCompare(CaseDTO caze) {
```

Comment:
caze? lol

Reply:
just following convention!

Reply:
As discussed. It's a Collection but its not a Collection. The service method returns what purports to be a Collection which contains a set, which cannot be accessed by the client, only the latest element. Consider either making it a Collection<T> or rename to be LatestCaseHelperMunglerThingy and do not expose it to the client, instead just the single latest case